backend/copr_backend/background_worker_build.py

```diff
@@ -381,7 +383,30 @@ def _proctitle(self, text):
         self.log.debug("setting title: %s", text)
         self.setproctitle(text)
 
+    def _check_build_interrupted(self):
+        """
+        Should we interrupet a running worker?
+        """
+        return (self._check_failed_resalloc_ticket()
+                or self._cancel_task_check_request())
+
+    @ttl_cache(ttl=10*60)
+    def _check_failed_resalloc_ticket(self):
+        """
+        Did the resalloc ticket fail?
+        """
+        try:
+            # Resalloc server can be temporarily down
+            self.host.ticket.collect()
+        except socket.error:
+            self.log.error("Connection to resalloc failed, "
+                           "using cached ticket status")
```

Comment:
I'm still not sure this is right.  Is this a bug in the resalloc client?  since we use the survive_server_restart option, it should [never return this exception ](https://github.com/praiskup/resalloc/blob/7a46fcc269a9e38174786ca3fb365e96c0a46cbd/resalloc/client.py#L46-L48)

Reply:
It is possible that is bug in the resalloc client.

I submitted a build, waited until the `fedora-39-x86_64` chroot is in the `running` state. Then I did `docker-compose kill resalloc`, and then `socket.gaierror` exception appeared in the logs.

Adding this try-except block helped, and I started seeing the "Connection to resalloc failed" message instead.

If you want, we can drop the try-except block here, and fix it in resalloc.

Reply:
That's suspicious,
```
In [1]: import socket

In [2]: isinstance(socket.gaierror(), socket.error)
Out[2]: True
```
Are we eventually using the "survives_server_restart=True"?

Reply:
What do you think about https://github.com/praiskup/resalloc/pull/136 ?

Reply:
The resalloc PR helped the issue, so I dropped the try-except here.
Now it seems that `self.host.ticket.collect()` runs forever until the resalloc servers gets back up again. No exceptions are appearing.